hoplite-watch/src/main/kotlin/com/sksamuel/hoplite/watch/watchers/consul.kt

```diff
@@ -0,0 +1,13 @@
+package com.sksamuel.hoplite.watch.watchers
+
+import com.orbitz.consul.cache.KVCache
+import com.sksamuel.hoplite.watch.Watchable
+
+class ConsulWatcher(private val kvCache: KVCache) : Watchable {
+  override fun watch(callback: () -> Unit) {
+    kvCache.addListener {
+      callback()
+    }
+    kvCache.start()
```

Comment:
We could also accept a string key + client and handle the kvCache creation in this method, but this gives callers a little more flexibility.

Reply:
Should we be starting the cache?

Reply:
Good question, I think it depends:
- not starting it puts the impetus on the caller to start, and makes configuration a bit less simple
- starting could cause problems if the consumer stops it and we're not longer receiving events